### ISSUE TITLE
Problems about training on one mechine 8 A100

### ISSUE THREAD

**Issue by ailearnerkyh** [cure-lab/MagicDrive#62]
Hello, I try to run the training code on one mechine with 8 A100 GPUs, and the accelerate config is as follows:
```
  # real default
  compute_environment: LOCAL_MACHINE
  deepspeed_config: {}
  distributed_type: MULTI_GPU
  downcast_bf16: 'no'
  dynamo_config: {}
  dynamo_backend: 'no'
  fsdp_config: {}
  main_training_function: main
  megatron_lm_config: {}
  rdzv_backend: static
  same_network: true
  tpu_env: []
  tpu_use_cluster: false
  tpu_use_sudo: false
  use_cpu: false
  gpu_ids: all
  
  # should be changed by command line
  mixed_precision: fp16
  num_machines: 1
  num_processes: 8
  machine_rank: 0
  main_process_ip: 'null'
  main_process_port: 1234
```
And I use command 'accelerate launch --mixed_precision fp16 -- gpus_ids all --num_processes 8 tools/train.py +exp=rawbox_mv2.0t_0.4.4 runner=8gpus_t' to train the model.
But when the code is running, the training speed is very slow. It takes about 30min to handle 8 examples. I have noticed that sometimes all the GPUs' utils are 100%, sometimes 1 or 2 GPUs' util will change to 0%. Why will this happen? Could you suggest some possible solutions for me, thank you!

**Comment by flymin** [comment]
Please check #52 and use the map cache as stated in the readme.

**Comment by ailearnerkyh** [comment]
Thank you for your reply! I have tried use the map cache, but the speed of training doesn't increase, and is takes me 10min for the first step. In the training process, the speed of processing each step is different, some steps take more than ten minutes, while others only take 2 seconds. Is this a normal phenomenon? And I want to know approximately how many epochs need to be trained, 100 epochs would take too much time. Thank you!
![1ad94cf1c473d8bf32dbdcffbe310bc](https://github.com/user-attachments/assets/7d980e9b-e872-49c5-9778-31f34be9cf09)

**Comment by flymin** [comment]
The phenomenon you described is exactly the case when map cache is not available. Could you check again to make sure maps are loaded from cache file? By default, our code fall back to map calculations on any error with the cache file without prompting error messages. You have to check it manually.

**Comment by ailearnerkyh** [comment]
Thank you for your suggestion. I have checked the Nuscenes_cache.yaml and set `dataset_cache_file_tag:12Hz_interp` and `dataset_cache_dirname: nuscenes_map_aux_12Hz_interp` . I think maybe I skip the Option1 and Option2, and I just download all the released pre-generated annotations. In this case, the data structure is as follows:
![30d9d92bea140122597a69c74b9d884](https://github.com/user-attachments/assets/f19b9563-bc28-444e-866f-a9b275616a66)
Is it because there is no `nuscenes_map_aux` folder in my `../data/`? Can you tell me how to generate the files in this directory? I found that the `dataset_process_root` in `map_cache_gen.yaml` is `.../ data/nuscenes_mmdet3d-t-use-break/`， But since I skipped Option 1, I don't have this directory.

**Comment by flymin** [comment]
The cache files are too large to share. You need to generate them locally. Please find instructions in the readme.

**Comment by ailearnerkyh** [comment]
Thank you very much for your reply. I have already used map cache, and the INFO also indicates that `using data cache from: ../data/nuscenes_map_aux_12Hz_interp/train_12Hz_interp.h5` and `using data cache from: ../data/nuscenes_map_aux_12Hz_interp/val_12Hz_interp.h5`, I carefully checked the train.log, the cache file path is correct. When I use interp, It still takes 20 hours to train 5000steps according to `rawbox_mv2.0t_0.4.4.yaml`. Could you please provide me with a `train.log` trained with interp data? I would like to know if my config setting is wrong, thank you very much!

**Comment by flymin** [comment]
According to your speed, it takes about 2weeks for 80k step, which is acceptable. As long as your gpu usage keeps high, it is good enough for your sever.

**Comment by yzli-0910** [comment]
@flymin I was wondering if training on full dataset and full iterations is essential when validating the model? If it takes 13 days for each try, the model tuning would  take much time.

Did you try other training settings during the design phase of this research? Such as using subset of nuScenes or reducing training iterations? I would be appreciated if you could share some details.

**Comment by flymin** [comment]
I think you are working on video generation. Image generation shouldn't have such an issue.

We use subsets with few videos (less than 100) to perform overfitting experiments, to validate the effectiveness of control modules and the overall training flow. You may design your own for other purposes. In most cases, you should observe advantages in thousands of iterations, and full training is only needed for one run.

**Comment by yzli-0910** [comment]
Thank you very much for the speedy answer! I will take a try.

**Comment by yzli-0910** [comment]
@flymin Based on your suggestions, I have tried the subset training experiments (1/10 of full nuScenes data), and I observed the overfitting (performed well on training set), but poor performance when testing, such as changing the location of 3d bbox but getting results without the car generated inside the box. I was wondering how could you validate the effectiveness of control modules with sub training set.

Thank you for your patience and I am looking forward to the reply.

**Comment by flymin** [comment]
Sure, you cannot expect generalization ability when testing with overfitting experiments. Just compare the fitting loss and gain some hints through the convergence speed. Otherwise, you have to design your own experimental settings.
